target/linux/generic/config-5.10

```diff
@@ -3523,7 +3523,9 @@ CONFIG_MODULE_UNLOAD=y
 # CONFIG_MPLS is not set
 # CONFIG_MPLS_IPTUNNEL is not set
 # CONFIG_MPLS_ROUTING is not set
-# CONFIG_MPTCP is not set
+CONFIG_MPTCP=y
+CONFIG_INET_MPTCP_DIAG=m
```

Comment:
building modesl here is useless. If you need the module please create a kmod package in package/kernel/linux/modules/

Reply:
The `CONFIG_INET_MPTCP_DIAG` option should better go to `kmod-inet-diag` in `package/kernel/linux/modules/netsupport.mk` Then you can also package the .ko file there.

Reply:
Not all kernels (like 5.4) support multipath TCP protocol，Modifying netsupport.mk needs to add the kernel version limit, but I don't know how to add it

Reply:
> Not all kernels (like 5.4) support

Does not matter here (on master branch): Only 5.10 and 5.15 kernels, but, for example, if you did not want your package to build on 5.10, you could use: `DEPEND+=@!LINUX_5.10`. `FILES`, `KCONFIG`, and `AUTOLOAD` support `setting@ge5.10`: with the `@` and following, setting would only apply greater or equal to kernel 5.10. also available: le, lt, gt

> I don't know how to add it

Have a look at KernelPackage/fs-cifs for a real example of kernel version conditionals in a kernel module package: https://github.com/openwrt/openwrt/blob/96aa052c406d460de8c4c1c3cf0eb9a900ae09c5/package/kernel/linux/modules/fs.mk#L86

Also, please do not open new PRs for the same topic: https://github.com/openwrt/openwrt/pull/11071. Normally if you want a new feature in older branches, it needs to get into master first.